src/pal/src/numa/numa.cpp

```diff
@@ -140,71 +142,91 @@ BOOL
 NUMASupportInitialize()
 {
 #if HAVE_NUMA_H
-    if (numa_available() != -1)
+    numaHandle = dlopen("libnuma.so", RTLD_LAZY);
+    if (numaHandle == 0)
     {
-        struct bitmask *mask = numa_allocate_cpumask();
-        int numaNodesCount = numa_max_node() + 1;
-
-        g_possibleCpuCount = numa_num_possible_cpus();
-        g_cpuCount = 0;
-        g_groupCount = 0;
+        numaHandle = dlopen("libnuma.so.1", RTLD_LAZY);
+    }
+    else if (numaHandle != 0)
```

Comment:
@janvorli, thanks for the review. I have made the fixes and re-pushed the changes. Can you please take a look?